### PR TITLE
IGNITE-11970 Fixed cleanup of CQ entry buffers.

### PR DESCRIPTION
**Pull request by petrov-mg**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/continuous/ContinuousQueryBuffersCleanupTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            }
+
+            assertTrue(cqListenerNotifiedLatch.await(getTestTimeout(), MILLISECONDS));
+
+            assertTrue(waitForCondition(() -> G.allGrids().stream().allMatch(this::isContinuesQueryBufferEmtpy), getTestTimeout()));
+        }
+    }
+
+    /** */
+    private boolean isContinuesQueryBufferEmtpy(Ignite ignite) {
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/continuous/ContinuousQueryBuffersCleanupTest.java:185]: Type: Emtpy -> Empty

File: modules/core/src/test/java/org/apache/ignite/internal/processors/continuous/ContinuousQueryBuffersCleanupTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+        contProc.localRoutineInfos().forEach((routineId, locRoutineInfo) ->
+            cqHandlers.add((CacheContinuousQueryHandler<?, ?>)locRoutineInfo.handler()));
+
+        contProc.clientRoutineInfos().forEach((nodeId, rmtRoutineInfos) ->
+            rmtRoutineInfos.forEach((routineId, locRoutineInfo) ->
+                cqHandlers.add((CacheContinuousQueryHandler<?, ?>)locRoutineInfo.handler())));
+
+        for (CacheContinuousQueryHandler<?, ?> cqHnd : cqHandlers) {
+            ConcurrentMap<Integer, CacheContinuousQueryEventBuffer> entryBufs = U.field(cqHnd, "entryBufs");
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/continuous/ContinuousQueryBuffersCleanupTest.java:201]: I don't like that a lot of reflection used by this test. In the same time you rename some variable and use it name in reflection, it's fragile logic. I know that there are two different packages for continuous queries and you need to access internal class fields in both packages and can't use package private methods. But, perhaps, we can introduce some class like `ContinuousQueriesTestUtils` in the package `org.apache.ignite.internal.processors.cache.query.continuous` end expose some methods to use in tests. WDYT?

Comment (petrov-mg) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/continuous/ContinuousQueryBuffersCleanupTest.java]: Fixed.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/continuous/ContinuousQueryBuffersCleanupTest.java (hunk 1 above)
Comment (petrov-mg) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/continuous/ContinuousQueryBuffersCleanupTest.java]: Fixed.

**Review by alex-plekhanov — APPROVED** [review]

**Pull request merged**